### PR TITLE
Add autogeneration of api docs

### LINKED ISSUE #305: Autogeneration of docs
I feel like we should have auto-generation for a lot of this or just do something like the GitOps Toolkit docs to avoid to do extra work whenever there are API changes. 

https://fluxcd.io/docs/components/kustomize/kustomization/

_Originally posted by @jonathan-innis in https://github.com/Azure/orkestra/issues/261#issuecomment-846626048_

### PR DESCRIPTION
**Pull request by mahalrs**
Closes #305

This change enables autogeneration of API docs. To generate docs, run the following `make` target:
```shell
make api-docs
```

And this will create/overwrite `./docs/api.md` file.

### CODE REVIEW

**Comment by mahalrs** [comment]
@nitishm The changed `api.md` file is automatically generated. Let me know if everything looks good.

File: hack/api-docs/config.json [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    {
+      "typeMatchPrefix": "^github.com/fluxcd/helm-controller/api/v2beta1",
+      "docsURLTemplate": "https://pkg.go.dev/github.com/fluxcd/helm-controller/api/v2beta1#{{ .TypeIdentifier }}"
+    }
+  ],
+  "typeDisplayNamePrefixOverrides": {
+    "k8s.io/api/": "Kubernetes ",
+    "k8s.io/apimachinery/pkg/apis/": "Kubernetes ",
+    "k8s.io/apiextensions-apiserver/": "Kubernetes ",
+    "github.com/fluxcd/helm-controller/api/": "helm-conntroler "
```
Comment (nitishm) [inline hack/api-docs/config.json:31]: Typo s/helm-conntroler/helm-controller/

**Review by nitishm — CHANGES_REQUESTED** [review]
Typo fix

File: hack/api-docs/config.json (hunk 1 above)
Comment (mahalrs) [reply hack/api-docs/config.json]: Done

File: Makefile [hunk 2]
```diff
@@ -61,6 +61,10 @@ manifests: controller-gen
 	$(CONTROLLER_GEN) $(CRD_OPTIONS) rbac:roleName=manager-role webhook paths="./..." output:crd:artifacts:config=config/crd/bases
 	$(CONTROLLER_GEN) $(CRD_OPTIONS) rbac:roleName=manager-role webhook paths="./..." output:crd:artifacts:config=chart/orkestra/crds
 
+# Generate API reference documentation
+api-docs: gen-crd-api-reference-docs
```
Comment (jonathan-innis) [inline Makefile:65]: @nitishm Let's create an issue for adding docs generation to CI

**Review by jonathan-innis — APPROVED** [review]
🚀  Great work @mahalrs 👏

File: docs/api.md [hunk 3]
```diff
@@ -3,103 +3,776 @@ layout: default
 title: API Reference
```
Comment (jonathan-innis) [inline docs/api.md:3]: Re-gen the docs after spelling fix, helm-controller is still misspelled

**Review by jonathan-innis — CHANGES_REQUESTED** [review]

File: docs/api.md (hunk 3 above)
Comment (mahalrs) [reply docs/api.md:3]: Fixed

**Review by jonathan-innis — APPROVED** [review]

**Review by nitishm — APPROVED** [review]
🚀 🎸

**Pull request merged**